NBitcoin/UInt2561.cs

```diff
@@ -288,6 +286,26 @@ public bool Equals(uint256 other)
 			return equals;
 		}
 
+		public int CompareTo(uint256 other)
+		{
+			if (other == null)
+			{
+				return 1;
+			}
+
+			return Comparison(this, other);
+		}
+
+		public int CompareTo(object obj)
+		{
+			if (obj != null && obj.GetType() != GetType())

```

Comment:
Please, do this 1 line:
```csharp
return obj is uint256 v ? CompareTo(v) : throw new ArgumentException($"Object is not an instance of uint256", nameof(obj));
```

Reply:
The problems is it will violate documentation  https://docs.microsoft.com/en-us/dotnet/api/system.icomparable.compareto?view=netframework-4.8#System_IComparable_CompareTo_System_Object_

Reply:
how does it violate it?

Reply:
Here is how it say:

```
The parameter, obj, must be the same type as the class or value type that implements this interface; otherwise, an ArgumentException is thrown.
```

Reply:
Yes, and my code is checking this.

Reply:
This handle the null case:
```csharp
return obj is uint256 v ? CompareTo(v) : 
       obj is null ? CompareTo(null as uint256) : throw new ArgumentException($"Object is not an instance of uint256", nameof(obj));
```

Reply:
The problems is if `obj` is a derived type it will not throw. I have another idea. The idea is making `uint256` class as sealed. But this is a breaking change if someone already derived from it.

Reply:
`uint256` can safely be sealed.

Reply:
We can seal `uint256`, or just don't care about derived type. This is such a corner case...